src/emacs/lean-info.el

```diff
@@ -67,6 +67,26 @@
       (lambda (&key record)
         (funcall cont record))))))
 
+(defun lean-get-info-record-at-point-sync ()
+  "Get info-record at point synchronously."
+  (with-demoted-errors "lean get info: %s"
+    (lean-server-send-synchronous-command
+     'info (list :file_name (buffer-file-name)
+                 :line (line-number-at-pos)
+                 :column (lean-line-offset)))))
+
+(defun lean-info-right-click-find-definition ()
+  "Offer to jump to definition of right-click target."
+  (interactive)
+  (let* ((info-record (plist-get (lean-get-info-record-at-point-sync) :record))
```

Comment:
> Emacs should do better than VS Code where possible :-)

Nothing beats friendly competition.  I hope emacs catches up, in particular with auto-completion and  elisp.  It always feels a bit like going back to the stone age, with no real module system, only manual namespacing, and no type-checker.

> [...] hole commands. [...] what's the good of knowing that a "Find definition" command exists that will throw an error?

Speaking of invalid commands, we currently show all hole commands for every hole--even if they wouldn't work.  Ideally I'd also like to have at most one replacement for every valid hole command.  That is, I think we should change `hole_command` to:
```lean
meta def hole_command :=
list pexpr -> tactic (list (string × tactic (option string)))
```
The nested `tactic string` is there to allow lazy computation of a replacement (in case it's expensive).  When executing a hole command in the editor, we'd first compute the list of possible replacements.  Then the user makes a single choice, and we evaluate the `tactic (option string)` (which could fail, I just want to get rid of the "obviously" impossible actions).